scapy/themes.py

```diff
@@ -288,11 +288,18 @@ class HTMLTheme2(HTMLTheme):
     style_right = "#[#span class=right#]#%s#[#/span#]#"
 
 
-def apply_ipython_color(shell):
+def apply_ipython_style(shell):
     """Updates the specified IPython console shell with
     the conf.color_theme scapy theme."""
     from IPython.terminal.prompts import Prompts, Token
-    shell.highlighting_style_overrides = {
+    from scapy.config import conf
+    class ClassicPrompt(Prompts):
+        def in_prompt_tokens(self, cli=None):
+           return [(Token.Prompt, conf.prompt),]
+        def out_prompt_tokens(self):
+           return [(Token.OutPrompt, ''),]
+    shell.prompts_class=ClassicPrompt # Set classic prompt style
+    shell.highlighting_style_overrides = { # Register and apply scapy color style
```

Comment:
It could be interesting to allow the definition of a Prompt class for conf.prompt:

```
    if isinstance(conf.prompt, basestring):
        class ClassicPrompt(Prompts):
            def in_prompt_tokens(self, cli=None):
                return [(Token.Prompt, conf.prompt),]
            def out_prompt_tokens(self):
                return [(Token.OutPrompt, ''),]
        shell.prompts_class = ClassicPrompt # Set classic prompt style
    else:
        shell.prompts_class = conf.prompt
```

Reply:
Sounds good